### ISSUE TITLE
ListView.builder with items in the bottom of the scolled area doesn't work

### ISSUE THREAD

**Issue by PashaTurok** [flutter/flutter#109983]
I need a ListView.builder with messages at the bottom.

[![enter image description here][1]][1]

I found two solution to do it:

**Solution # 1 with SingleChildScrollView**

    class Home extends StatelessWidget {
      @override
      Widget build(BuildContext context) {
        return Scaffold(
            appBar: AppBar(
              title: Text('AppBar'),
            ),
            body: LayoutBuilder(builder: (context, constraints) {
              return SingleChildScrollView(
                  child: ConstrainedBox(
                      constraints: BoxConstraints(minWidth: constraints.maxWidth, minHeight: constraints.maxHeight),
                      child: IntrinsicHeight(
                        child: Column(
                            mainAxisSize: MainAxisSize.max,
                            children: [
                              Text('header'),
                              Expanded(
                                child: Container(
                                  color: Colors.green,
                                  child: Text('body'),
                                ),
                              ),
                              Text('footer'),
                               // MY CODE STARTS
                               ListView.builder(
                                 itemCount: 2,
                                 itemBuilder: (context, index) {
                                   return Text("Message " + index.toString());
                                 }
                               )
                               // MY CODE ENDS
                            ]
                        ),
                      )
                  )
              );
            })
        );
      }
    }

This the result I get:

    The following assertion was thrown during performLayout():
    RenderViewport does not support returning intrinsic dimensions.
    
    Calculating the intrinsic dimensions would require instantiating every child of the viewport, which defeats the point of viewports being lazy.
    
    If you are merely trying to shrink-wrap the viewport in the main axis direction, consider a RenderShrinkWrappingViewport render object (ShrinkWrappingViewport widget), which achieves that effect without implementing the intrinsic dimension API.

**Solution # 2 with CustomScrollView**

    class Home extends StatelessWidget {
      @override
      Widget build(BuildContext context) {
        return Scaffold(
            appBar: AppBar(
              title: Text('AppBar'),
            ),
            body: SafeArea(
            child: CustomScrollView(
              slivers: <Widget>[
                SliverFillRemaining(
                  hasScrollBody: false,
                  // fillOverscroll: true, // Set true to change overscroll behavior. Purely preference.
                  child: Align(
                    alignment: Alignment.bottomCenter,
                     // MY CODE STARTS
                    child:  ListView.builder(
                        itemCount: 2,
                        itemBuilder: (context, index) {
                          return Text("Message " + index.toString());
                        }
                    ),
                   // MY CODE ENDS
                  ),
                )
              ],
            ),
          )
        );
      }
    }

 And this is the result:
```
The following assertion was thrown during performLayout():
RenderViewport does not support returning intrinsic dimensions.

Calculating the intrinsic dimensions would require instantiating every child of the viewport, which defeats the point of viewports being lazy.

If you are merely trying to shrink-wrap the viewport in the main axis direction, consider a RenderShrinkWrappingViewport render object (ShrinkWrappingViewport widget), which achieves that effect without implementing the intrinsic dimension API.
```
I think this is a bug, because both solutions work when instead of `ListView.builder` I use `Text` widget . So, there mustn't be a problem if instead of one `Text` I have multiple same `Text` inside list. 

  [1]: https://i.stack.imgur.com/P7dkZ.png
  [2]: https://stackoverflow.com/a/66820328/5057736

**Comment by maheshmnj** [comment]
Hi @PashaTurok, Thanks for filing the issue. If I understand it right you want your list to start from bottom of the screen and turn into the scrollable list when it exceeds screen height is that correct?

**Comment by maheshmnj** [comment]
@PashaTurok  The issue with your second code is you are passing `scrollBody:false` and passing a scrollable widget (ListView.builder), setting it to true (default) should fix the issue.

https://user-images.githubusercontent.com/31410839/186085314-2c165d37-1a8b-46da-825c-36829db4723c.mov



<details>
<summary>code sample</summary>

```dart
import "package:flutter/material.dart";

void main() {
  runApp(const MaterialApp(home: Scaffold(body: MyApp())));
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Home(),
    );
  }
}

class Home extends StatefulWidget {
  @override
  State<Home> createState() => _HomeState();
}

class _HomeState extends State<Home> {
  void addMessage() {
    list.add('message $counter');
    setState(() {});
  }

  List<String> list = [];

  int counter = 0;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        floatingActionButton: FloatingActionButton(
          onPressed: addMessage,
          child: Icon(Icons.add),
        ),
        appBar: AppBar(
          title: Text('AppBar'),
        ),
        body: new CustomScrollView(
          slivers: <Widget>[
            SliverFillRemaining(
              child: Align(
                alignment: Alignment.bottomCenter,
                child: ListView.builder(
                    itemCount: list.length,
                    reverse: true,
                    itemBuilder: (context, index) {
                      return Text("Message " + index.toString());
                    }),
                // MY CODE ENDS
              ),
            )
          ],
        ));
  }
}

```
</details> 


<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.4 21F79 darwin-arm, locale en-IN)
    • Flutter version 3.0.5 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (5 days ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.44.0

[✓] Connected device (3 available)
    • sdk gphone arm64 (mobile) • emulator-5554 • android-arm64  • Android 11 (API 30) (emulator)
    • macOS (desktop)           • macos         • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)              • chrome        • web-javascript • Google Chrome 103.0.5060.114

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

Closing as this is working as intended, In case you disagree feel free to write in the comments and we will reopen it.

Thank you.

**Comment by maheshmnj** [comment]
@PashaTurok for your point 
1. You could reverse your list too, it is completely dependent on your implementation, My example shared was mainly to show that you could implement a list that starts from the bottom of the screen.
2.  Can you be more specific and share a code specifying the expected vs actual result?

**Comment by iapicca** [comment]
> > Can you be more specific and share a code specifying the expected vs actual result?

> I really have nothing to add. I described the problem in details.

@PashaTurok 
I share @maheshmnj opinion, there aren't enough information to call it a bug (maybe it is, maybe it isn't)
opening a [new issue](https://github.com/flutter/flutter/issues/new?assignees=&labels=&template=2_bug.md&title=)
you are prompted to provide some information like

```
**Expected results:** <!-- what did you want to see? -->

**Actual results:** <!-- what did you see? -->
```

if you think that it's a bug and not just a problem with your implementation
I think the first part is quite important to be answered

(also [this is not how it works](https://github.com/flutter/flutter/issues/110341))

**Comment by iapicca** [comment]
>  I need to align my messages at the bottom of the scrollable area. As I found out it is not such a simple task in flutter. 

I find it a very simple task if it was just that, 
but it seams to me that what you are trying to achieve something different; 
honestly I'm confused about what that would be:

###
In your `solution #1` the `ListView`  is inside a `SingleChildScrollView` obviously won't even compile w/o set height
and if you provide a height the `ListView` will scroll independently from the `SingleChildScrollView`
(see [code sample](https://dartpad.dev/?id=eadcecbe0065313e2efea11495052300))
###
In your `solution #2` all the and seems that you just need a list of widget that starts from the bottom, 
if that's the case it's a very simple task (see [code sample](https://dartpad.dev/?id=d8a8cb24e191ee7b164529c332235c44))

I would add a 3rd option where the you scroll the whole page and just add elements to the list below (see [code sample](https://dartpad.dev/?id=b385102d8de1c636e4c2b12e0a54acda))

Bottom line
I don't think this is an implementation issue or even a programming issue, 
this seems to me a design problem or rather a design decision that still need to be made.
**Definitely not a flutter issue**

**Comment by iapicca** [comment]
>  The only problem NO ONE OF YOUR SOLUTIONS DOES WHAT I NEED.
> No need to answer. I just waste my time. No one even understands the problem.

Are you reporting a bug or you just need help with your app?
Flutter community is great, but if you are asking "how do I implement XYZ" 
in a repo where usually developers report bugs
ain't gonna fly

if you need help to implement something take a look [here](https://flutter.dev/community)
I recommend the discord channel!
